### DISCUSSION TITLE
Disable unneeded plugins & Improve compile time

### DISCUSSION THREAD

**Discussion by DelinWorks** [axmolengine/axmol#640] [Q&A] [answered]
Hi! how can I disable plugins that I don't need? like `DragonSpine` and `FairyGui` etc
The reason I want them disabled is whenever I change something in the engine's header files I have to recompile the entire engine AS WELL as plugins that take even longer to compile.

Or maybe disabling plugins isn't a solution, is it normal that when I change a single line of code in a header `H` file of the engine I get to compile the entire thing again, this doesn't happen when modifying source files `CPP`

Any help would be much appreciated! :)

**Comment by halx99** [comment, marked as answer]
see https://github.com/adxeproject/adxe/blob/dev/CMakeOptions.md

**Reply by DelinWorks** [reply]
@halx99 There is another problem... when I add preprocessor macros to engine's project like this:
![image](https://user-images.githubusercontent.com/45469625/169635979-cb750d7f-c848-405c-bd5b-eb9dd2edfc83.png)
Every time I rebuild cmake these macros get removed! is there a way I can add them through cmake?
I've heard of `add_compile_definitions()` but didn't exactly know how to use it :>

**Reply by halx99** [reply]
These macros is cmake options, you shoud pass them as command line  to cmake when generate native project
